cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginClass.java

```diff
@@ -39,11 +38,11 @@
   private final String configFieldName;
   private final Map<String, PluginPropertyField> properties;
   private final Set<String> endpoints;
-  private final Set<String> requirements;
+  private final Requisites requisites;
```

Comment:
why not name it Requirements? That seems to be the most natural name given the annotation is named that.

Reply:
I see it's because you want to use the same object in the Provisioner. But I think that's confusing, because the set of datasets here is the set of required datasets, but the set the provisioner returns is the set of supported datasets. I think it's fine to have two classes - Requirements and Capabilities